### DISCUSSION TITLE
Declarative shadow DOM rendering is broken

### DISCUSSION THREAD

**Discussion by AlexandreBonaventure** [livewire/livewire#8321] [7. Bugs (archived)]
### Livewire version

3.4.10

### Laravel version

v11

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

1. Open the repro
2. See the devtool warning message: shadowroot is not initialized properly 

<img width="1024" alt="Capture d’écran, le 2024-04-17 à 12 20 55" src="https://github.com/livewire/livewire/assets/4596409/d3fd789d-79ea-4789-a59d-6adb2ee94791">

### Repository or Wirebox URL

https://wirebox.app/b/y87vq

### Additional information

The issue is directly visible when using lazy loading for instance because it needs to morph the element. 
Now, morphing is using innerHTML which does not support declarative shadow DOM:
https://github.com/livewire/livewire/blob/7c3757b0f1e56425dbd89f8173ddc3517dfa1dde/js/morph.js#L13

DSD support is pretty recent so that's not really a surprise but Livewire users could really benefit from it (easy encapsulation, server-rendered web components, etc..)
https://developer.chrome.com/docs/css-ui/declarative-shadow-dom#parser-only

With all that said, there is a new browser API to replace innerHTML with DSD support 
https://caniuse.com/mdn-api_shadowroot_sethtmlunsafe
We could check if it's supported and fallback to innerHTML, crafting an MR atm.

ps: it looks like innerHTML could be deprecated in favor of https://developer.mozilla.org/en-US/docs/Web/API/Element/setHTML in a far future

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by AlexandreBonaventure** [comment]
MR: https://github.com/livewire/livewire/pull/8322

**Comment by joshhanley** [comment]
@AlexandreBonaventure can you make the wirebox link public?

**Reply by AlexandreBonaventure** [reply]
woops! my bad, should be fixed now! @joshhanley
